ppr-ui/src/views/discharge/ReviewRegistration.vue

```diff
@@ -0,0 +1,227 @@
+// local helpers/enums/interfaces/resources
+import { APIRegistrationTypes, RouteNames, UIRegistrationTypes } from '@/enums' // eslint-disable-line no-unused-vars
+import {
+  ActionBindingIF, FeeSummaryIF, ErrorIF, AddPartiesIF, // eslint-disable-line no-unused-vars
+  RegistrationTypeIF, AddCollateralIF, LengthTrustIF // eslint-disable-line no-unused-vars
+} from '@/interfaces'
+import { RegistrationTypes } from '@/resources'
+import { convertDate, getFeatureFlag, getFinancingStatement } from '@/utils'
+import { StatusCodes } from 'http-status-codes'
+
+@Component({
+  components: {
+    ButtonFooter,
+    RegistrationFee,
+    RegistrationLengthTrust,
+    Stepper,
+    Collateral,
+    DebtorSummary,
+    RegisteringPartySummary,
+    SecuredPartySummary,
+    FolioNumberSummary
+  }
+})
+export default class ReviewRegistration extends Vue {
+  @Getter getRegistrationType: RegistrationTypeIF
+
+  @Action setAddCollateral: ActionBindingIF
+  @Action setAddSecuredPartiesAndDebtors: ActionBindingIF
+  @Action setFeeSummary: ActionBindingIF
+  @Action setLengthTrust: ActionBindingIF
+  @Action setRegistrationType: ActionBindingIF
+
+  /** Whether App is ready. */
+  @Prop({ default: false })
+  private appReady: boolean
+
+  @Prop({ default: false })
+  private isJestRunning: boolean
+
+  private dataLoaded = false
+  private financingStatementDate: Date = null
+
+  private get asOfDateTime (): string {
+    // return formatted date
+    if (this.financingStatementDate) {
+      return `${convertDate(this.financingStatementDate, true, true)}`
+    }
+    return ''
+  }
+
+  private get isAuthenticated (): boolean {
+    return Boolean(sessionStorage.getItem(SessionStorageKeys.KeyCloakToken))
+  }
+
+  // the number of the registration being discharged
+  private get registrationNumber (): string {
+    return this.$route.query['reg-num'] as string || ''
+  }
+
+  private get registrationTypeUI (): UIRegistrationTypes {
+    return this.getRegistrationType?.registrationTypeUI || null
+  }
+
+  private get registrationType (): APIRegistrationTypes {
+    return this.getRegistrationType?.registrationTypeAPI || null
+  }
+
+  private async loadRegistration (): Promise<void> {
+    if (!this.registrationNumber) {
+      console.error('No registration number given to discharge. Redirecting to dashboard...')
+      this.$router.push({
+        name: RouteNames.DASHBOARD
+      })
+      return
+    }
+    this.financingStatementDate = new Date()
+    const financingStatement = await getFinancingStatement(true, this.registrationNumber)
+    if (financingStatement.error) {
+      this.emitError(financingStatement.error)
+    } else {
+      // load data into the store
+      const registrationType = RegistrationTypes.find(
+        (reg, index) => {
+          if (reg.registrationTypeAPI === financingStatement.type) {
+            return true
+          }
+        })
+      // FUTURE: get general collateral from response
```

Comment:
why future?

Reply:
I don't think you were there, but they need to redesign general collateral so we aren't including it until the design is finished

Reply:
gotcha